### PR TITLE
Add possibility to publish batches of events

### PR DESCRIPTION
**Pull request by johanssontobias**

### CODE REVIEW

File: knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs [hunk 1]
```diff
@@ -94,12 +98,25 @@ public ServiceBus(IServiceBusConfiguration config)
             await SendAsync(client, brokeredMessage, cancellationToken).ConfigureAwait(false);
         }
 
+        public async Task PublishEventsAsync<T>(IList<T> messages, CancellationToken cancellationToken = default) where T : IServiceBusEvent
+        {
+            var client = await ClientFactory.GetSenderClient<T>().ConfigureAwait(false);
+            if (!messages.Any()) return;
```
Comment (NiklasArbin) [inline knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs:104]: Maybe do this first before fetching the sender client

File: knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs [hunk 2]
```diff
@@ -94,12 +98,25 @@ public ServiceBus(IServiceBusConfiguration config)
...
+            var brokeredMessages = new List<ServiceBusMessage>();
+            foreach (var message in messages)
+            {
+                brokeredMessages.Add(await CreateMessageAsync(message).ConfigureAwait(false));
```
Comment (NiklasArbin) [inline knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs:108]: You will need to handle exceptions here if the message gets too large, or leave it up to every client

File: knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs (hunk 1 above)
Comment (johanssontobias) [reply knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs:103]: It now returns before fetching a client on both batch PublishEventsAsync and batch SendAsync.

File: knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs (hunk 2 above)
Comment (johanssontobias) [reply knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs:108]: Not currently done in batch SendAsync either. Perhaps a future feature request?

Comment (NiklasArbin) [reply knightbus-azureservicebus/src/KnightBus.Azure.ServiceBus/ServiceBus.cs:108]: Change send batch async also

**Pull request closed without merging**